nx_cugraph/algorithms/lowest_common_ancestors.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cupy as cp
+import networkx as nx
+import numpy as np
+import pylibcugraph as plc
+
+from nx_cugraph.convert import _to_directed_graph
+from nx_cugraph.utils import (
+    _groupby,
+    index_dtype,
+    networkx_algorithm,
+    not_implemented_for,
+)
+
+__all__ = ["lowest_common_ancestor"]
+
+
+@not_implemented_for("undirected")
+@networkx_algorithm(is_incomplete=True, version_added="24.12", _plc="bfs")
+def lowest_common_ancestor(G, node1, node2, default=None):
+    """May not always raise NetworkXError for graphs that are not DAGs."""
+    G = _to_directed_graph(G)
+
+    # if not nxcg.is_directed_acyclic_graph(G):  # TODO
```

Comment:
Do we need to wait on this before merging?

Reply:
It would be nice to, but I don't think it's super important in practice so I would say no. We add a note to the docstring. If/when PLC can help answer whether a graph is a DAG, then we should add `nxcg.is_directed_acyclic_graph` ASAP.

The difference is that if the graph is not a DAG, then we may still give an answer or we may raise as networkx does. It's common to know whether your graph is a DAG or not (often by construction).